### PR TITLE
Match deduct_current_move_fp

### PR DESCRIPTION
**Pull request by JoshDuMan**

### CODE REVIEW

**Comment by JoshDuMan** [comment]
https://decomp.me/scratch/7Pi18

**Comment by BowserSlug** [comment]
🚀 This PR matches 1 function (+0.00%) on `us`.

**Review by nanaian — APPROVED** [review]

File: src/190B20.c [hunk 1]
```diff
@@ -1157,9 +1157,48 @@ s32 count_power_plus(s32 arg0) {
...
+            actor->unk_204[0] = 0;
+        }
 
-INCLUDE_ASM(s32, "190B20", reset_actor_turn_info);
+    }
+    actor = battleStatus->playerActor;
+    actor->hpChangeCounter = 0;
+    actor->damageCounter = 0;
+    actor->unk_204[0] = 0;
+    actor = battleStatus->partnerActor;
```
Comment (nanaian) [inline src/190B20.c:1195]:
```suggestion

    actor = battleStatus->playerActor;
    actor->hpChangeCounter = 0;
    actor->damageCounter = 0;
    actor->unk_204[0] = 0;
    
    actor = battleStatus->partnerActor;
```

**Review by nanaian — COMMENTED** [review]
Actually I'd spread these lines out a lil bit based on which actor is being reset

File: src/190B20.c [hunk 2]
```diff
@@ -1157,9 +1157,49 @@ s32 count_power_plus(s32 arg0) {
...
+
+    if (fpCost != 0) {
+        fpCost -= player_team_is_ability_active(actor, ABILITY_FLOWER_SAVER);
+        fpCost -= player_team_is_ability_active(actor, ABILITY_FLOWER_FANATIC) * 2;
+        if (fpCost < 1) {
+            fpCost = 1;
+        }
+    }
+
+    playerData->curFP = playerData->curFP - fpCost;
```
Comment (ethteck) [inline src/190B20.c:1174]: -=

File: src/190B20.c [hunk 3]
```diff
@@ -1157,9 +1157,49 @@ s32 count_power_plus(s32 arg0) {
...
+        actor = battleStatus->enemyActors[i];
+        if (actor != NULL) {
+            actor->hpChangeCounter = 0;
+            actor->damageCounter = 0;
+            actor->unk_204[0] = 0;
+        }
 
-INCLUDE_ASM(s32, "190B20", reset_actor_turn_info);
+    }
+    actor = battleStatus->playerActor;
```
Comment (ethteck) [inline src/190B20.c:1191]: Do you actually have to keep saving the current actor into this temp? You can't just do battleStatus->playerActor-> whatever = 0?

**Review by ethteck — CHANGES_REQUESTED** [review]

File: src/190B20.c (hunk 3 above)
Comment (ethteck) [reply src/190B20.c:1191]: nm. tried it and you can't.

**Pull request merged**